src/validation/validator-name.ts

```diff
@@ -36,12 +36,34 @@ export default class ValidatorName implements IValidator {
   }
 
   private titleCase(value: string): string {
-    const words = value.toLowerCase().split(' ');
-    const titleCase = (word: string) => word[0].toUpperCase() + word.slice(1);
-    const isRomanNumeral = /^[ivx]+$/ig;
-    const titleCased = words
+    if (!value) {
+      return '';
+    }
+
+    const titleCase = (word: string) => word[0]?.toUpperCase() + word.slice(1);
+    const isRomanNumeral = /^[ivx]+$/i;
+
+    const processWord = (word: string) => {
+      if (word.match(isRomanNumeral)) {
+        return word.toUpperCase();
+      }
+      return word.includes('-')
```

Comment:
here we are splitting on a symbol, calling `processWord` for each word, and then rejoining. we are doing this again on line 62-65.  

what do you think of putting that pattern into a function `splitAndProcess(value, ' ')` and `splitAndProcess(value, '-')`?

Reply:
Good catch.